### PR TITLE
[CR]Corpse item types

### PR DESCRIPTION
**Pull request by Coolthulhu**
How do I see #20069

Basically:
* Corpse item type data is generated for every monster type (that has normal death) on finalization
* Corpse is an `itype` slot
* Monsters can drop predefined corpses (not used here)
* Can wish for corpses

### CODE REVIEW

**Comment by Coolthulhu** [comment]
It looks huge, but actually isn't.
I had to move item finalization in item factory to functions, so that runtime items can be finalized. This accounts for most of the lines changed because of indentation.

**Comment by pisskop** [comment]
So one could define the spawning of a corpse of a monster, it would spawn, and any effects/properties it held will be retained?  such as reviving or burning?

**Comment by Coolthulhu** [comment]
Yes, I moved setting up those properties to the constructor.
The biggest change is that you can fully customize the properties of a corpse and use it in recipes, mission targets and so on.

**Comment by BrettDong** [comment]
Is `revive_msg` and `revive_carried_msg` customizable in JSON? If so, those two new entries need to be added to `lang/extract_json_strings.py`.

File: src/item.cpp [hunk 1]
```diff
@@ -180,14 +184,10 @@ item item::make_corpse( const mtype_id& mt, int turn, const std::string &name )
         debugmsg( "tried to make a corpse with an invalid mtype id" );
     }
 
-    item result( "corpse", turn >= 0 ? turn : int( calendar::turn ) );
+    item result( mt.obj().get_corpse_itype(), turn >= 0 ? turn : int( calendar::turn ) );
```
Comment (codemime) [inline src/item.cpp]: `string_id<>` overloads the arrow operator. I'd use it instead of `obj()` for the sake of readability: `mt->get_corpse_itype()`

**Comment by codemime** [comment]
> This accounts for most of the lines changed because of indentation.

It would be better to astyle in separate PRs. Makes it easier to see what's actually changed.

File: src/item_factory.cpp [hunk 2]
```diff
@@ -92,251 +95,279 @@ static bool assign_coverage_from_json( JsonObject &jo, const std::string &key,
...
+    }
+    // for ammo not specifying loudness (or an explicit zero) derive value from other properties
+    if( obj.ammo ) {
+        if( obj.ammo->loudness < 0 ) {
+            obj.ammo->loudness = ( obj.ammo->damage + obj.ammo->pierce + obj.ammo->range ) * 2;
+        }
+
+        const auto &mats = obj.materials;
+        if( std::find( mats.begin(), mats.end(), material_id( "hydrocarbons" ) ) == mats.end() &&
+            std::find( mats.begin(), mats.end(), material_id( "oil" ) ) == mats.end() ) {
```
Comment (codemime) [inline src/item_factory.cpp]: There's `item::made_of()` which does that job.

**Comment by codemime** [comment]
Looks good to go, but has minor merge conflicts.

**Pull request closed without merging**